component/org.wso2.carbon.identity.dpop/README.md

```diff
@@ -0,0 +1,78 @@
+# DPoP component
+
+DPoP ( Demonstrating Proof of Possession ) is an additional security mechanism for the token
+generation which overcomes the issue of bearer token which will not validate between who is
+requested token and who is actually using the token for the access of a particular resource.The specification defines a mechanism to prevent illegal API calls from succeeding only with a stolen access token. In the traditional mechanism, API access is allowed only if the access token presented by the client application is valid
+
+## Specification 
+https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop-02
+
+## Design 
+
+### Sequence Diagram.
+
+1. DPoP token request
+![Screenshot from 2021-10-25 23-06-12](https://user-images.githubusercontent.com/26603378/138743329-5cc54271-08a6-44ec-938e-d675bdd24717.png)
+
+
+2. Invoking protected APIs with DPoP token and DPoP proof.
+![Invoke API(2)](https://user-images.githubusercontent.com/26603378/138742776-3d2c2714-c87e-4f77-9dce-24fde3df600e.jpeg)
+
+### Sample client application to create dpop proof
+PR : [wso2 /samples-is#302 ](https://github.com/wso2/samples-is/pull/302 )
+
+### Deployment Instructions
+
+1. Build the project using mvn clean install.
+2. Add the  org.wso2.carbon.identity.dpop-2.4.3-SNAPSHOT.jar JAR into the <IS_HOME>/repository/components/droppings folder.
+3. Add the below configuration to deployment.toml file.
+
+ ```
+[[event_listener]]
+id = "dpop_listener"
+type = "org.wso2.carbon.identity.core.handler.AbstractIdentityHandler"
+name="org.wso2.carbon.identity.dpop.listener.OauthDPoPInterceptorHandlerProxy"
+order = 13
+enable = true
+properties.header_validity_period = 90
+
+[[oauth.custom_token_validator]]
+type = "dpop"
+class = "org.wso2.carbon.identity.dpop.validators.DPoPTokenValidator"
+
+[oauth.grant_type.uma_ticket]
```

Comment:
Can you explain the reason for using uma config here?

Reply:
In order to enable the custom introspection data provider.